### PR TITLE
[docs] Update system integrators guide links

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Updates faucet links and some typos

### CODE REVIEW

**Review by 0xmaayan — APPROVED** [review]

File: developer-docs-site/docs/guides/system-integrators-guide.md [hunk 1]
```diff
@@ -62,16 +62,16 @@ Either of these methods will expose a [REST API service](./aptos-apis.md) at `ht
       <li>REST API: <a href="https://fullnode.testnet.aptoslabs.com">https://fullnode.testnet.aptoslabs.com</a></li>
       <li><a href="https://fullnode.testnet.aptoslabs.com/v1/spec#/">REST API Spec</a></li>
       <li>Indexer API: <a href="https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql">https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql</a></li>
-      <li><a href="https://cloud.hasura.io/public/graphiql?endpoint=https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql">Indexer Graphiql</a></li>
-      <li><a href="https://aptoslabs.com/testnet-faucet">Faucet Site</a></li>
+      <li><a href="https://cloud.hasura.io/public/graphiql?endpoint=https://indexer-testnet.staging.gcp.aptosdev.com/v1/graphql">Indexer GraphQL</a></li>
+      <li>Faucet API: <a href="https://faucet.devnet.aptoslabs.com">https://faucet.testnet.aptoslabs.com</a></li>
```
Comment (banool) [inline developer-docs-site/docs/guides/system-integrators-guide.md:66]: The URL says testnet but you're actually pointing to devnet.

**Review by banool — DISMISSED** [review]

**Review by saharct — APPROVED** [review]
@gregnazario I made some minor changes and fixed the hyperlink that @banool pointed out.

File: developer-docs-site/docs/guides/system-integrators-guide.md (hunk 1 above)
Comment (saharct) [reply developer-docs-site/docs/guides/system-integrators-guide.md]: I went ahead and fixed the link.

**Review by banool — APPROVED** [review]
Looks good!

**Pull request merged**